{
  boolean caughtEx=false;
  try {
    try {
      mscFunctionInvoker.invoke();
    }
 catch (    RuntimeException e) {
      String exStackString=ExceptionUtils.getStackTrace(e);
      assertTrue("Verifying this exception came after authorization check",exStackString.contains("org.apache.hadoop.hive.metastore.ObjectStore"));
    }
    assertFalse("Authz Exception should have been thrown in remote mode",isRemoteMetastoreMode);
    System.err.println("No auth exception thrown");
  }
 catch (  MetaException e) {
    System.err.println("Caught exception");
    caughtEx=true;
    assertTrue(e.getMessage().contains(MetaStoreAuthzAPIAuthorizerEmbedOnly.errMsg));
  }
  if (!isRemoteMetastoreMode) {
    assertFalse("No exception should be thrown in embedded mode",caughtEx);
  }
}
